### PR TITLE
[SPARK-31133][SQL][DOC] fix sql ref doc for DML

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

`INSERT OVERWRITE DIRECTORY` can only use file format (class implements `org.apache.spark.sql.execution.datasources.FileFormat`). This PR fixes it and other minor improvement.

### CODE REVIEW

File: docs/sql-ref-syntax-dml-load.md
Comment (cloud-fan) [inline docs/sql-ref-syntax-dml-load.md]: ``LOAD DATA` only works for hive table

File: docs/sql-ref-syntax-dml-load.md
Comment (cloud-fan) [inline docs/sql-ref-syntax-dml-load.md]: This must be a hive table

File: docs/sql-ref-syntax-dml-load.md
Comment (cloud-fan) [inline docs/sql-ref-syntax-dml-load.md]: This is used as the source so doesn't need to be a hive table.

**Comment by cloud-fan** [comment]
cc @dongjoon-hyun @maropu @viirya

File: docs/sql-ref-syntax-dml-insert-overwrite-directory.md
Comment (maropu) [inline docs/sql-ref-syntax-dml-insert-overwrite-directory.md]: nit: `Spark file format` -> `a given Spark file format`?

**Review by maropu — APPROVED** [review]
LGTM except for one minor comment.

File: docs/sql-ref-syntax-dml-load.md
Comment (dongjoon-hyun) [reply docs/sql-ref-syntax-dml-load.md]: Hi, @cloud-fan . This reminds me.
```
SPARK-30098 Use default datasource as provider for CREATE TABLE syntax
```

@marmbrus and @gatorsmile . Do we need to revert SPARK-30098 due to its silent behavior change?

Also, cc @rxin since he is the release manager for 3.0.0.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Shall we wait for the decision for SPARK-30098?

File: docs/sql-ref-syntax-dml-load.md
Comment (dongjoon-hyun) [reply docs/sql-ref-syntax-dml-load.md]: I filed SPARK-31136 to track the discussion and the final result.

Comment (cloud-fan) [reply docs/sql-ref-syntax-dml-load.md]: The cost of break is mostly users can't run `LOAD TABLE`?

Comment (dongjoon-hyun) [reply docs/sql-ref-syntax-dml-load.md]: Please don't get me wrong. You know that I loved that patch. However, the new policy is designed to ban this kind of behavior change (SPARK-30098). Technically,
1. The benefit is just saving a few word `USING PARQUET` or something.
2. The downside is breaking the existing user pipelines.

Comment (dongjoon-hyun) [reply docs/sql-ref-syntax-dml-load.md]: I updated the JIRA description with @cloud-fan 's example.

Comment (cloud-fan) [reply docs/sql-ref-syntax-dml-load.md]: > The benefit is just saving a few word USING PARQUET or something.

This is a completely underestimate. Please see my comment in https://github.com/apache/spark/pull/27894#issuecomment-598534428

File: docs/sql-ref-syntax-dml-insert-overwrite-directory.md
Comment (viirya) [inline docs/sql-ref-syntax-dml-insert-overwrite-directory.md]: Isn't AVRO also valid? Btw, does DELTA implement FileFormat? If it means DeltaDataSource, looks like it only implements DataSourceRegister.

File: docs/sql-ref-syntax-dml-load.md
Comment (viirya) [reply docs/sql-ref-syntax-dml-load.md]: good catch!

**Review by viirya — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Hi, All. I marked SPARK-31136 as a correctness issue and added the example.

File: docs/sql-ref-syntax-dml-insert-overwrite-directory.md
Comment (cloud-fan) [reply docs/sql-ref-syntax-dml-insert-overwrite-directory.md]: Avro implements `FileFormat`. I don't know if DELTA is supported in `INSERT OVERWRITE [ LOCAL ] DIRECTORY`, let me remove it as it's an external source.

**Comment by cloud-fan** [comment]
shall we merge this PR first? It correctly describes the current behavior. If we want to change the behavior, we should update the document accordingly, instead of blocking this PR and wait for it.

**Comment by cloud-fan** [comment]
I'll merge it in a few days if no objection. The CHAR/VARCHAR discussion is still going on in dev list.

**Comment by cloud-fan** [comment]
merging to master/3.0, thanks for review!

**Pull request closed without merging**
